addons/delivery/models/stock_picking.py

```diff
@@ -187,6 +220,18 @@ def check_packages_are_identical(self):
                 raise UserError(_('You are shipping different packaging types in the same shipment.\nPackaging Types: %s' % package_names))
         return True
 
+    @api.multi
+    @api.returns('self', lambda value: value.id)
+    def message_post(self, **kwargs):
+        """ Overrides mail_thread message_post so that we can set is_sent_delivery_mail to true  of boolean field value to identify email 
+        is sent and make send confirmation email button normal from primary.
+        """
+        if self.env.context.get('default_model') == 'stock.picking' and self.env.context.get('default_res_id') and self.env.context.get('mark_delivery_as_sent'):
```

Comment:
Whut ? I bet you can simplify this code ... hint: override and inheritance mechanism ;) .